### PR TITLE
gnome-builder: add ctags

### PR DESCRIPTION
**Pull request by worldofpeace**
Gnome Builder uses Ctags for autocompletion.

From looking at [this](https://gitlab.gnome.org/GNOME/gnome-builder/blob/master/data/gsettings/meson.build#L7) adding it as a build input should be sufficient.

cc @jameysharp  @jtojnar 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by xeji** [comment]
@GrahamcOfBorg build gnome-builder

**Comment by worldofpeace** [comment]
@jtojnar Using direnv or nix-shell for ctags doesn't make sense.

You can find errors in the output when it can't find it a soon as you open a file.

There's even an option to set the [path](https://gitlab.gnome.org/GNOME/gnome-builder/blob/master/meson_options.txt#L19). This is more of a dependency I think than something for someone's specific dev environment.

**Comment by jameysharp** [comment]
I find this argument about ctags convincing, and it's also possible I was wrong to argue to the contrary for other build-time dependencies.

Perhaps the rule should be this: anything that upstream gnome-builder tries to detect while running the meson configure script should be a packaging dependency here.

My fear was about trying to track down all gnome-builder's subprocess invocations that assumed that at runtime various tools would be on `$PATH`. I was worried that trying to locally patch all of those references would be a maintenance problem. But that doesn't apply to ctags in this case.

It's also possible that some of those run-time dependencies should be turned into build-time dependencies. If it really only makes sense to use one version of a tool on a given platform, I'd expect that upstream would accept patches doing so, and then there's no need to maintain the large patches I was worried about here in nixpkgs.

In short: @jtojnar, my suggestion would be to go ahead and merge this pull request, and we can think about whether this reasoning applies to other tools later.
